devices/power.py

```diff
@@ -299,6 +312,123 @@ def reset(self):
 	time.sleep(5)
 	self.on()
 
+class Ip9820(PowerDevice):
+    def __init__(self, ip_address, port):
+        self.ip_address = ip_address
+        self.port = port
+        self.oid_Outlet = '1.3.6.1.4.1.38107.1.3'
+        
+    def on(self):
+       oid = self.oid_Outlet + '.' + str(self.port) + '.0'
+       print("set Ip9820 port on")
+       Snmp("set",self.ip_address,"private",oid ,'i',1)
+       result = Snmp("get",self.ip_address,"private",oid)
+       print("result = %s" % result)
+       return result
+    
+    def off(self):
+        oid = self.oid_Outlet + '.' + str(self.port) + '.0'
+        print("set Ip9820 port off")
+        Snmp("set",self.ip_address,"private",oid ,'i',0)
+        result = Snmp("get",self.ip_address,"private",oid)
+        print("result = %s" % result)
+        return result
+    
+    def reset(self):
+	self.off()
+	time.sleep(5)
+	self.on()
+    
+    def getPowerStatus(self):
+        oid = self.oid_Outlet + '.' + str(self.port) + '.0'
+        result = Snmp("get",self.wan_ip,"private",oid)
+        return result
+    
+    def getPowerStatusAll(self):
+        oid = self.oid_Outlet
+        result = Snmp("walk",self.wan_ip,"private",oid)
```

Comment:
I see you are using some SNMP commands here actually, but does the same API that the other IP board use work instead?